### ISSUE TITLE
Cable Status sensor is unreliable since last update

### ISSUE THREAD

**Issue by flagshipenis** [mattrayner/pod-point-home-assistant-component#60]
I had based all of my charging automations off this sensor because it was the quickest to update and most reliable.
Now this sensor can go an entire charging session without changing state from "unplugged" even if status is "charging" and it sais it is connected to the cloud

**Comment by Sahnia** [comment]
I agree the sensor is now unreliable, constantly changing it's status. 

![Screenshot_20240526-185027](https://github.com/mattrayner/pod-point-home-assistant-component/assets/33041410/d54c4159-be54-44fe-b8b1-a9b8a5246fe8)

**Comment by mattrayner** [comment]
I will take a look and try to recreate, I've not seen this on my pod. Its coming into the school half term this week so it will be unlikely that I will be able to take a look until next week. Was there a certain time that it started happening? Did you restart your Home Assistant instance before seeing it?

Anything to help me re-create it would be awesome

**Comment by Sahnia** [comment]
Hi,

I have restarted home assistant a number of times and the problem continues. I am not sure how to recreate it. 

Completely understand about half term. For me this issue has been on going for a while and not particularly problematic so not urgent. 

Sorry I cannot be more helpful.

**Comment by flagshipenis** [comment]
I need to clarify. The sensor also reports "connected" when it is not actually connected some times too. It is totally useless at this state.
This started happening after the first update that made PSL status report "pending" and the other new states.
I hope this helps

**Comment by willenic** [comment]
I also have this issue.

**Comment by btxluke** [comment]
Another pod point with this issue. Mine in combination with a eCorsa, which only updates it's cable sensor when Stellantis can be arsed.

**Comment by btxluke** [comment]
![image](https://github.com/user-attachments/assets/513faca7-777d-4b59-8919-4a9497ddd4fc)

It seems to me, that what is happening here is that when you switch  "Charging Allowed" the cable status immediately changes to "unplugged" regardless of the actual plug status (is it renegotiating with the car?), and stays that way for about 5 minutes.

I use predbat to plan my carging slots, it refreshes every 5 minutes, so i was getting stuck in a doom loop of charging / not charging creating the behaviour you see above.

As a quick nasty fix, I created a helper that toggles according to the initial plug status, and then toggles off if the cable is reported unplugged for more than 30 minutes, and updated predbat to make decisions based on the smoothed sensor. It works, but feels nasty.

**Comment by btxluke** [comment]
Final update:

This confirms it, I think:

![image](https://github.com/user-attachments/assets/e8dc9ee3-5925-444a-9c4e-007284615480)

Everytime "Charging Allowed" is switched, Cable Status is reset for +/- 5 Minutes. If you are doing your charging programmatically, you'll need to make a sensor to smooth it out.

**Comment by flagshipenis** [comment]
> Final update:
> 
> This confirms it, I think:
> 
> ![image](https://private-user-images.githubusercontent.com/95568985/399454277-e8dc9ee3-5925-444a-9c4e-007284615480.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4Mzk2MDgsIm5iZiI6MTczNTgzOTMwOCwicGF0aCI6Ii85NTU2ODk4NS8zOTk0NTQyNzctZThkYzllZTMtNTkyNS00NDRhLTljNGUtMDA3Mjg0NjE1NDgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAyVDE3MzUwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExZTAzOWQ1ODkzNTkzNWM4ZGVhMjNiYmUxOTUxNTUyMzcxYjQ1ZjI3Zjg3Y2E3ZGVmNTJhYjM3MGVlOGRiNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EvZfaUVzJiB5kMpKH-HUxPViER-uhMcJNdc01MMotOo)
> 
> Everytime "Charging Allowed" is switched, Cable Status is reset for +/- 5 Minutes. If you are doing your charging programmatically, you'll need to make a sensor to smooth it out.

**Comment by flagshipenis** [comment]
Interesting! I did not see this at all when i was troubble shooting.

I will have a look at it again!

Thank you 🙂
